core/modules/date/date.theme.inc

```diff
@@ -474,4 +474,223 @@ function template_preprocess_date_views_filter_form(&$vars) {
   $vars['description'] = backdrop_render($form['description']) . backdrop_render($form);
 }
 
+/**
+ * Jump in and move the pager.
+ */
+function date_preprocess_views_view(&$vars) {
+  $view = $vars['view'];
+  if (!empty($view->date_info) && !empty($view->date_info->date_pager_position)) {
+    switch ($view->date_info->date_pager_position) {
+      case 'top':
+        $vars['header'] .= $vars['pager'];
+        $vars['pager'] = '';
+        break;
+
+      case 'both':
+        $vars['header'] .= $vars['pager'];
+        break;
+
+      default:
+        // Already on the bottom.
+    }
+  }
+}
+
+/**
+ * Preprocess function for Date pager template.
+ */
+function template_preprocess_date_views_pager(&$vars) {
```

Comment:
All instances of `$vars` should be replaced with `$variables` as per our coding standards. It looks like this code may be left over from Drupal 6.